sdk/src/Core/Amazon.Util/EC2InstanceMetadata.cs

```diff
@@ -480,6 +458,72 @@ public static string IdentityPkcs7
             }
         }
 
+        /// <summary>
+        /// Returns the temporary security credentials (AccessKeyId, SecretAccessKey, SessionToken, and Expiration) 
+        /// associated with the IAM roles on the instance.
+        /// </summary>
+        /// <returns>Temporary security credentials</returns>
+        public static IDictionary<string, IAMSecurityCredentialMetadata> GetIAMSecurityCredentials()
+        {
+            var list = GetItems("/iam/security-credentials");
+            if (list == null)
```

Comment:
Since the only thing different between these sync and async is the first method. Can you put the code after getting the list of credentials in a separate private method that takes in the list and then reuse the code between the methods.

Reply:
We are also using individual items and making another sync/async call `var json = await GetDataAsync("/iam/security-credentials/" + item).ConfigureAwait(false);`.

I have refactored code that deserializes credential data into separate method.

Reply:
Fair point, I missed the `GetData` and `GetDataAsync` calls.